pkg/promlib/querydata/request.go

```diff
@@ -234,9 +234,14 @@ func (s *QueryData) fetch(traceCtx context.Context, client *client.Client, q *mo
 func (s *QueryData) rangeQuery(ctx context.Context, c *client.Client, q *models.Query, enablePrometheusDataplaneFlag bool) backend.DataResponse {
 	res, err := c.QueryRange(ctx, q)
 	if err != nil {
+		return addErrorSourceToDataResponse(err)
+	}
+
+	if res.StatusCode/100 != 2 {
+		// for differentiating the source of http errors by status code
 		return backend.DataResponse{
-			Error:  err,
-			Status: backend.StatusBadGateway,
+			Error:       err,
```

Comment:
I think that in this case error might be `nil` and you should check the response in what form are we getting the error message. This is what we use in Loki https://github.com/grafana/grafana/blob/e86283e8b173b328c4710720b8f807f359f57a77/pkg/tsdb/loki/api.go#L145  and https://github.com/grafana/grafana/blob/e86283e8b173b328c4710720b8f807f359f57a77/pkg/tsdb/loki/api.go#L126